sdk/core/core/inc/az_http.h

```diff
@@ -33,6 +33,12 @@ typedef enum
  */
 typedef az_span az_http_method;
 
+/**
+ * @brief Define az_http_headers as an az_span so it is limited to that type and not to any az_span
+ *
+ */
+typedef az_span az_http_headers;
```

Comment:
az_http_headers is only used internally, and is located in _internal struct - if you still want to go with the typedef, I think we should add an underscore `typedef az_span _az_http_headers;`

Reply:
That's a good point, I'll switch to that

Reply:
Are you also going to move the function to internal header?